File: Backend/OrderItem.py
# Backend/OrderItem.py
from .CartItem import CartItem

class OrderItem:
    """
    Immutable snapshot of a Cart Item for creating a record on a order.
    """
    def __init__(self, cartItem: CartItem):
        self.product = cartItem.product
        self.quantity = cartItem.quantity
        self.unit_price = cartItem.product.price
        self.total_price = cartItem.subtotal

File: Backend/src/Store/OrderItem.java
package Store;

/**
 * Immutable detail of a CartItem for an Order.
 */
public class OrderItem {
    private final Product product;
    private final int quantity;
    private final double unitPrice;
    private final double totalPrice;

    public OrderItem(CartItem ci) {
        this.product    = ci.getProduct();
        this.quantity   = ci.getQuantity();
        this.unitPrice  = product.getPrice();
        this.totalPrice = ci.getSubtotal();
    }

    public Product getProduct()   { return product;    }
    public int     getQuantity()  { return quantity;   }
    public double  getUnitPrice() { return unitPrice;  }
    public double  getTotalPrice(){ return totalPrice; }
}
